### PR TITLE
Add function to easily generate preset pass managers

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a new function, `generate_preset_pass_managers()`, which
is used to quickly generate the preset pass manager for manual
interaction. Prior to this new function there were lower level functions
which are used to generate the individual pass manager objects for each
optimization level but they are hidden behind an abstraction object,
`PassManagerConfig` which is the only input. This makes it harder to work
with because the user has to manually remember to generate the config
object prior to creating the pass manager. Similarly this new function
simplifies the creation of preset pass managers by having a single entry
point instead of 4 for each optimization level.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
We should make sure that this doesn't get included in the release without https://github.com/Qiskit/qiskit-terra/pull/8109. If #8109 doesn't get included this function won't work with `BackendV2` instances.

File: qiskit/transpiler/preset_passmanagers/__init__.py [hunk 1]
```diff
@@ -20,13 +20,180 @@
...
+        PassManager: The preset pass manager for the given options
+
+    Raises:
+        ValueError: if an invalid value for ``optimization_level`` is passed in.
+    """
+
+    if backend is not None:
+        pm_config = PassManagerConfig.from_backend(
+            backend,
+            target=target,
```
Comment (Cryoris) [inline qiskit/transpiler/preset_passmanagers/__init__.py:144]: This is a suggestion you're very welcome to ignore, but would this be a bit less prone to forgetting an argument in either of the if-clauses if we'd write it something like the following?
```python
settings = {"target": target, "basis_gates": basis_gates, ...}
if backend is not None:
    pm_config = PassManagerConfig.from_backend(backend, **settings)
else:
    pm_config = PassManagerConfig(**settings)
```

File: test/python/transpiler/test_preset_passmanagers.py [hunk 2]
```diff
@@ -940,3 +941,15 @@ def test_size_optimization(self, level):
...
         self.assertLess(circ.size(), qc.size())
         self.assertLessEqual(circ.depth(), qc.depth())
+
+
+@ddt
+class TestGeenratePresetPassManagers(QiskitTestCase):
+    """Test generate_preset_pass_manager function."""
+
+    @data(0, 1, 2, 3)
+    def test_with_backend(self, optimization_level):
```
Comment (Cryoris) [inline test/python/transpiler/test_preset_passmanagers.py:951]: Do we want to add a test where the backend is `None` so we cover both cases of the if-clause in the function?

File: qiskit/transpiler/preset_passmanagers/__init__.py (hunk 1 above)
Comment (mtreinish) [reply qiskit/transpiler/preset_passmanagers/__init__.py]: Good call, done in: https://github.com/Qiskit/qiskit-terra/pull/8125/commits/4e86d95a2324e36932b48bb12206d421a511542e

File: test/python/transpiler/test_preset_passmanagers.py (hunk 2 above)
Comment (mtreinish) [reply test/python/transpiler/test_preset_passmanagers.py:952]: Yep, that's a good idea. Done in: https://github.com/Qiskit/qiskit-terra/pull/8125/commits/c40a0f127f2f69384ee852824029b9db1feba88e

File: releasenotes/notes/generate_pass_manager_preset-1e6c9641accd5d60.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+features:
+  - |
+    Added a new function :func:`qiskit.transpiler.generate_preset_pass_manager` which can
```
Comment (Cryoris) [inline releasenotes/notes/generate_pass_manager_preset-1e6c9641accd5d60.yaml:4]: Should we also add a reno for `target_to_backend_properties` which is now public?

**Review by Cryoris — COMMENTED** [review]
Overall LGTM! I left some picky comments on the docstring (since we're touching it anyways, why not 😛) and do we want to add tests for `target_to_backend_properties`? Maybe that's not necessary since it's implicitly tested in the `test_with_no_backend` test 🙂

File: releasenotes/notes/generate_pass_manager_preset-1e6c9641accd5d60.yaml (hunk 3 above)
Comment (mtreinish) [reply releasenotes/notes/generate_pass_manager_preset-1e6c9641accd5d60.yaml]: I really didn't want to advertise it too much. It's public because I'm using across multiple modules and it felt weird to do otherwise but I'm trying to move everything away from the `BackendProperties` object. 

As for the test coverage we have implicit coverage already via tests that use `transpile()`. We've had test coverage for that since #7227 I can add dedicated tests for though if you'd prefer to have that since it is a standalone function now.

Comment (Cryoris) [reply releasenotes/notes/generate_pass_manager_preset-1e6c9641accd5d60.yaml]: No that's fine with me, it's also not included in any `__init__` 👍🏻

**Review by Cryoris — APPROVED** [review]
